Networking/NetworkingTests/Remote/WCPayRemoteTests.swift

```diff
@@ -53,4 +53,269 @@ final class WCPayRemoteTests: XCTestCase {
 
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
+
+    /// Verifies that loadAccount properly handles the nominal response. We'll also validate the
+    /// statement descriptor, currencies and country here.
+    ///
+    func test_loadAccount_properly_returns_account() {
+        let remote = WCPayRemote(network: network)
+
+        network.simulateResponse(requestUrlSuffix: "payments/accounts", filename: "wcpay-account-complete")
+
+        // When
+        let result: Result<WCPayAccount, Error> = waitFor { promise in
+            remote.loadAccount(for: self.sampleSiteID) { result in
+                promise(result)
+            }
+        }
+
+        // Then
+        switch result {
+        case .success(let account):
+            XCTAssertEqual(account.status, .complete)
+            XCTAssertEqual(account.statementDescriptor, "MY.FANCY.US.STORE")
+            XCTAssertEqual(account.defaultCurrency, "usd")
+            XCTAssertEqual(account.supportedCurrencies, ["usd"])
+            XCTAssertEqual(account.country, "US")
+        case .failure(let error):
+            XCTAssertNil(error)
+        }
```

Comment:
One way we can make this more intentional is:

1. Answer “What is the expected status of `result`”? Successful or failure? 
2. Get the actual success value and test that. 

In this way, we don't need to check for `.failure`, which clarifies that we only expected the result to be successful. Here's what I think that can look like: 

```suggestion
        XCTAssertTrue(result.isSuccess)

        let account = try result.get()
        XCTAssertEqual(account.status, .complete)
        XCTAssertEqual(account.statementDescriptor, "MY.FANCY.US.STORE")
        XCTAssertEqual(account.defaultCurrency, "usd")
        XCTAssertEqual(account.supportedCurrencies, ["usd"])
        XCTAssertEqual(account.country, "US")
```

P.S. We'd need to make the test function `throws`. 😄

Reply:
Good idea - opened a new issue in https://github.com/woocommerce/woocommerce-ios/issues/3900